README.md
# swagger-java-client

Cloudflare API
- API version: 4.0.0
  - Build date: 2023-02-22T13:40:27.702461541+09:00[Asia/Tokyo]

Interact with Cloudflare's products and services via the Cloudflare API. For more information on how to create and troubleshoot API tokens, refer to our [API fundamentals](https://developers.cloudflare.com/fundamentals/api/).  Totally new to Cloudflare? [Start here](https://developers.cloudflare.com/fundamentals/get-started/).


*Automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen)*


## Requirements

Building the API client library requires:
1. Java 1.7+
2. Maven/Gradle

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>io.swagger</groupId>
  <artifactId>swagger-java-client</artifactId>
  <version>1.0.0</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
compile "io.swagger:swagger-java-client:1.0.0"
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

* `target/swagger-java-client-1.0.0.jar`
* `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        Object body = null; // Object | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            InlineResponse20025 result = apiInstance.accessApplicationsAddABookmarkApplication(body, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsAddABookmarkApplication");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        SchemasAppId appId = new SchemasAppId(); // SchemasAppId | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            AppsComponentsSchemasIdResponse result = apiInstance.accessApplicationsDeleteAnAccessApplication(appId, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsDeleteAnAccessApplication");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        SchemasAppId appId = new SchemasAppId(); // SchemasAppId | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            AppsComponentsSchemasSingleResponse result = apiInstance.accessApplicationsGetAnAccessApplication(appId, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsGetAnAccessApplication");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            AppsComponentsSchemasResponseCollection result = apiInstance.accessApplicationsListAccessApplications(identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsListAccessApplications");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        SchemasAppId appId = new SchemasAppId(); // SchemasAppId | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            ComponentsSchemasEmptyResponse result = apiInstance.accessApplicationsRevokeServiceTokens(appId, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsRevokeServiceTokens");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        SchemasAppId appId = new SchemasAppId(); // SchemasAppId | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            PolicyCheckResponse result = apiInstance.accessApplicationsTestAccessPolicies(appId, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsTestAccessPolicies");
            e.printStackTrace();
        }
    }
}
import io.swagger.client.*;
import io.swagger.client.auth.*;
import io.swagger.client.model.*;
import io.swagger.client.api.AccessApplicationsApi;

import java.io.File;
import java.util.*;

public class AccessApplicationsApiExample {

    public static void main(String[] args) {
        ApiClient defaultClient = Configuration.getDefaultApiClient();

        // Configure API key authorization: api_email
        ApiKeyAuth api_email = (ApiKeyAuth) defaultClient.getAuthentication("api_email");
        api_email.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_email.setApiKeyPrefix("Token");

        // Configure API key authorization: api_key
        ApiKeyAuth api_key = (ApiKeyAuth) defaultClient.getAuthentication("api_key");
        api_key.setApiKey("YOUR API KEY");
        // Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
        //api_key.setApiKeyPrefix("Token");


        AccessApplicationsApi apiInstance = new AccessApplicationsApi();
        Object body = null; // Object | 
        SchemasAppId appId = new SchemasAppId(); // SchemasAppId | 
        AppsComponentsSchemasIdentifier identifier = new AppsComponentsSchemasIdentifier(); // AppsComponentsSchemasIdentifier | 
        try {
            InlineResponse20025 result = apiInstance.accessApplicationsUpdateABookmarkApplication(body, appId, identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AccessApplicationsApi#accessApplicationsUpdateABookmarkApplication");
            e.printStackTrace();
        }
    }
}
```

## Documentation for API Endpoints

All URIs are relative to *https://api.cloudflare.com/client/v4*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AccessApplicationsApi* | [**accessApplicationsAddABookmarkApplication**](docs/AccessApplicationsApi.md#accessApplicationsAddABookmarkApplication) | **POST** /accounts/{identifier}/access/apps | Add a Bookmark application
*AccessApplicationsApi* | [**accessApplicationsDeleteAnAccessApplication**](docs/AccessApplicationsApi.md#accessApplicationsDeleteAnAccessApplication) | **DELETE** /accounts/{identifier}/access/apps/{app_id} | Delete an Access application
*AccessApplicationsApi* | [**accessApplicationsGetAnAccessApplication**](docs/AccessApplicationsApi.md#accessApplicationsGetAnAccessApplication) | **GET** /accounts/{identifier}/access/apps/{app_id} | Get an Access application
*AccessApplicationsApi* | [**accessApplicationsListAccessApplications**](docs/AccessApplicationsApi.md#accessApplicationsListAccessApplications) | **GET** /accounts/{identifier}/access/apps | List Access applications
*AccessApplicationsApi* | [**accessApplicationsRevokeServiceTokens**](docs/AccessApplicationsApi.md#accessApplicationsRevokeServiceTokens) | **POST** /accounts/{identifier}/access/apps/{app_id}/revoke_tokens | Revoke service tokens
*AccessApplicationsApi* | [**accessApplicationsTestAccessPolicies**](docs/AccessApplicationsApi.md#accessApplicationsTestAccessPolicies) | **GET** /accounts/{identifier}/access/apps/{app_id}/user_policy_checks | Test Access policies
*AccessApplicationsApi* | [**accessApplicationsUpdateABookmarkApplication**](docs/AccessApplicationsApi.md#accessApplicationsUpdateABookmarkApplication) | **PUT** /accounts/{identifier}/access/apps/{app_id} | Update a Bookmark application
*AccessAuthenticationLogsApi* | [**accessAuthenticationLogsGetAccessAuthenticationLogs**](docs/AccessAuthenticationLogsApi.md#accessAuthenticationLogsGetAccessAuthenticationLogs) | **GET** /accounts/{identifier}/access/logs/access_requests | Get Access authentication logs
*AccessBookmarkApplicationsDeprecatedApi* | [**accessBookmarkApplicationsDeprecatedCreateABookmarkApplication**](docs/AccessBookmarkApplicationsDeprecatedApi.md#accessBookmarkApplicationsDeprecatedCreateABookmarkApplication) | **POST** /accounts/{identifier}/access/bookmarks/{uuid} | Create a Bookmark application
*AccessBookmarkApplicationsDeprecatedApi* | [**accessBookmarkApplicationsDeprecatedDeleteABookmarkApplication**](docs/AccessBookmarkApplicationsDeprecatedApi.md#accessBookmarkApplicationsDeprecatedDeleteABookmarkApplication) | **DELETE** /accounts/{identifier}/access/bookmarks/{uuid} | Delete a Bookmark application
*AccessBookmarkApplicationsDeprecatedApi* | [**accessBookmarkApplicationsDeprecatedGetABookmarkApplication**](docs/AccessBookmarkApplicationsDeprecatedApi.md#accessBookmarkApplicationsDeprecatedGetABookmarkApplication) | **GET** /accounts/{identifier}/access/bookmarks/{uuid} | Get a Bookmark application
*AccessBookmarkApplicationsDeprecatedApi* | [**accessBookmarkApplicationsDeprecatedListBookmarkApplications**](docs/AccessBookmarkApplicationsDeprecatedApi.md#accessBookmarkApplicationsDeprecatedListBookmarkApplications) | **GET** /accounts/{identifier}/access/bookmarks | List Bookmark applications
*AccessBookmarkApplicationsDeprecatedApi* | [**accessBookmarkApplicationsDeprecatedUpdateABookmarkApplication**](docs/AccessBookmarkApplicationsDeprecatedApi.md#accessBookmarkApplicationsDeprecatedUpdateABookmarkApplication) | **PUT** /accounts/{identifier}/access/bookmarks/{uuid} | Update a Bookmark application
*AccessGroupsApi* | [**accessGroupsCreateAnAccessGroup**](docs/AccessGroupsApi.md#accessGroupsCreateAnAccessGroup) | **POST** /accounts/{identifier}/access/groups | Create an Access group
*AccessGroupsApi* | [**accessGroupsDeleteAnAccessGroup**](docs/AccessGroupsApi.md#accessGroupsDeleteAnAccessGroup) | **DELETE** /accounts/{identifier}/access/groups/{uuid} | Delete an Access group
*AccessGroupsApi* | [**accessGroupsGetAnAccessGroup**](docs/AccessGroupsApi.md#accessGroupsGetAnAccessGroup) | **GET** /accounts/{identifier}/access/groups/{uuid} | Get an Access group
*AccessGroupsApi* | [**accessGroupsListAccessGroups**](docs/AccessGroupsApi.md#accessGroupsListAccessGroups) | **GET** /accounts/{identifier}/access/groups | List Access groups
*AccessGroupsApi* | [**accessGroupsUpdateAnAccessGroup**](docs/AccessGroupsApi.md#accessGroupsUpdateAnAccessGroup) | **PUT** /accounts/{identifier}/access/groups/{uuid} | Update an Access group
*AccessIdentityProvidersApi* | [**accessIdentityProvidersAddAnAccessIdentityProvider**](docs/AccessIdentityProvidersApi.md#accessIdentityProvidersAddAnAccessIdentityProvider) | **POST** /accounts/{identifier}/access/identity_providers | Add an Access identity provider
*AccessIdentityProvidersApi* | [**accessIdentityProvidersDeleteAnAccessIdentityProvider**](docs/AccessIdentityProvidersApi.md#accessIdentityProvidersDeleteAnAccessIdentityProvider) | **DELETE** /accounts/{identifier}/access/identity_providers/{uuid} | Delete an Access identity provider
*AccessIdentityProvidersApi* | [**accessIdentityProvidersGetAnAccessIdentityProvider**](docs/AccessIdentityProvidersApi.md#accessIdentityProvidersGetAnAccessIdentityProvider) | **GET** /accounts/{identifier}/access/identity_providers/{uuid} | Get an Access identity provider
*AccessIdentityProvidersApi* | [**accessIdentityProvidersListAccessIdentityProviders**](docs/AccessIdentityProvidersApi.md#accessIdentityProvidersListAccessIdentityProviders) | **GET** /accounts/{identifier}/access/identity_providers | List Access identity providers
*AccessIdentityProvidersApi* | [**accessIdentityProvidersUpdateAnAccessIdentityProvider**](docs/AccessIdentityProvidersApi.md#accessIdentityProvidersUpdateAnAccessIdentityProvider) | **PUT** /accounts/{identifier}/access/identity_providers/{uuid} | Update an Access identity provider
*AccessKeyConfigurationApi* | [**accessKeyConfigurationGetTheAccessKeyConfiguration**](docs/AccessKeyConfigurationApi.md#accessKeyConfigurationGetTheAccessKeyConfiguration) | **GET** /accounts/{identifier}/access/keys | Get the Access key configuration
*AccessKeyConfigurationApi* | [**accessKeyConfigurationRotateAccessKeys**](docs/AccessKeyConfigurationApi.md#accessKeyConfigurationRotateAccessKeys) | **POST** /accounts/{identifier}/access/keys/rotate | Rotate Access keys
*AccessKeyConfigurationApi* | [**accessKeyConfigurationUpdateTheAccessKeyConfiguration**](docs/AccessKeyConfigurationApi.md#accessKeyConfigurationUpdateTheAccessKeyConfiguration) | **PUT** /accounts/{identifier}/access/keys | Update the Access key configuration
*AccessMTlsAuthenticationApi* | [**accessMTlsAuthenticationAddAnMTlsCertificate**](docs/AccessMTlsAuthenticationApi.md#accessMTlsAuthenticationAddAnMTlsCertificate) | **POST** /accounts/{identifier}/access/certificates | Add an mTLS certificate
*AccessMTlsAuthenticationApi* | [**accessMTlsAuthenticationDeleteAnMTlsCertificate**](docs/AccessMTlsAuthenticationApi.md#accessMTlsAuthenticationDeleteAnMTlsCertificate) | **DELETE** /accounts/{identifier}/access/certificates/{uuid} | Delete an mTLS certificate
*AccessMTlsAuthenticationApi* | [**accessMTlsAuthenticationGetAnMTlsCertificate**](docs/AccessMTlsAuthenticationApi.md#accessMTlsAuthenticationGetAnMTlsCertificate) | **GET** /accounts/{identifier}/access/certificates/{uuid} | Get an mTLS certificate
*AccessMTlsAuthenticationApi* | [**accessMTlsAuthenticationListMTlsCertificates**](docs/AccessMTlsAuthenticationApi.md#accessMTlsAuthenticationListMTlsCertificates) | **GET** /accounts/{identifier}/access/certificates | List mTLS certificates
*AccessMTlsAuthenticationApi* | [**accessMTlsAuthenticationUpdateAnMTlsCertificate**](docs/AccessMTlsAuthenticationApi.md#accessMTlsAuthenticationUpdateAnMTlsCertificate) | **PUT** /accounts/{identifier}/access/certificates/{uuid} | Update an mTLS certificate
*AccessPoliciesApi* | [**accessPoliciesCreateAnAccessPolicy**](docs/AccessPoliciesApi.md#accessPoliciesCreateAnAccessPolicy) | **POST** /accounts/{identifier}/access/apps/{uuid}/policies | Create an Access policy
*AccessPoliciesApi* | [**accessPoliciesDeleteAnAccessPolicy**](docs/AccessPoliciesApi.md#accessPoliciesDeleteAnAccessPolicy) | **DELETE** /accounts/{identifier}/access/apps/{uuid1}/policies/{uuid} | Delete an Access policy
*AccessPoliciesApi* | [**accessPoliciesGetAnAccessPolicy**](docs/AccessPoliciesApi.md#accessPoliciesGetAnAccessPolicy) | **GET** /accounts/{identifier}/access/apps/{uuid1}/policies/{uuid} | Get an Access policy
*AccessPoliciesApi* | [**accessPoliciesListAccessPolicies**](docs/AccessPoliciesApi.md#accessPoliciesListAccessPolicies) | **GET** /accounts/{identifier}/access/apps/{uuid}/policies | List Access policies
*AccessPoliciesApi* | [**accessPoliciesUpdateAnAccessPolicy**](docs/AccessPoliciesApi.md#accessPoliciesUpdateAnAccessPolicy) | **PUT** /accounts/{identifier}/access/apps/{uuid1}/policies/{uuid} | Update an Access policy
*AccessServiceTokensApi* | [**accessServiceTokensCreateAServiceToken**](docs/AccessServiceTokensApi.md#accessServiceTokensCreateAServiceToken) | **POST** /accounts/{identifier}/access/service_tokens | Create a service token
*AccessServiceTokensApi* | [**accessServiceTokensDeleteAServiceToken**](docs/AccessServiceTokensApi.md#accessServiceTokensDeleteAServiceToken) | **DELETE** /accounts/{identifier}/access/service_tokens/{uuid} | Delete a service token
*AccessServiceTokensApi* | [**accessServiceTokensListServiceTokens**](docs/AccessServiceTokensApi.md#accessServiceTokensListServiceTokens) | **GET** /accounts/{identifier}/access/service_tokens | List service tokens
*AccessServiceTokensApi* | [**accessServiceTokensRefreshAServiceToken**](docs/AccessServiceTokensApi.md#accessServiceTokensRefreshAServiceToken) | **POST** /accounts/{identifier}/access/service_tokens/{uuid}/refresh | Refresh a service token
*AccessServiceTokensApi* | [**accessServiceTokensRotateAServiceToken**](docs/AccessServiceTokensApi.md#accessServiceTokensRotateAServiceToken) | **POST** /accounts/{identifier}/access/service_tokens/{uuid}/rotate | Rotate a service token
*AccessServiceTokensApi* | [**accessServiceTokensUpdateAServiceToken**](docs/AccessServiceTokensApi.md#accessServiceTokensUpdateAServiceToken) | **PUT** /accounts/{identifier}/access/service_tokens/{uuid} | Update a service token
*AccessShortLivedCertificateCasApi* | [**accessShortLivedCertificateCAsCreateAShortLivedCertificateCa**](docs/AccessShortLivedCertificateCasApi.md#accessShortLivedCertificateCAsCreateAShortLivedCertificateCa) | **POST** /accounts/{identifier}/access/apps/{uuid}/ca | Create a short-lived certificate CA
*AccessShortLivedCertificateCasApi* | [**accessShortLivedCertificateCAsDeleteAShortLivedCertificateCa**](docs/AccessShortLivedCertificateCasApi.md#accessShortLivedCertificateCAsDeleteAShortLivedCertificateCa) | **DELETE** /accounts/{identifier}/access/apps/{uuid}/ca | Delete a short-lived certificate CA
*AccessShortLivedCertificateCasApi* | [**accessShortLivedCertificateCAsGetAShortLivedCertificateCa**](docs/AccessShortLivedCertificateCasApi.md#accessShortLivedCertificateCAsGetAShortLivedCertificateCa) | **GET** /accounts/{identifier}/access/apps/{uuid}/ca | Get a short-lived certificate CA
*AccessShortLivedCertificateCasApi* | [**accessShortLivedCertificateCAsListShortLivedCertificateCAs**](docs/AccessShortLivedCertificateCasApi.md#accessShortLivedCertificateCAsListShortLivedCertificateCAs) | **GET** /accounts/{identifier}/access/apps/ca | List short-lived certificate CAs
*AccountBillingProfileApi* | [**accountBillingProfileBillingProfileDetails**](docs/AccountBillingProfileApi.md#accountBillingProfileBillingProfileDetails) | **GET** /accounts/{account_identifier}/billing/profile | Billing Profile Details
*AccountLevelCustomNameserversApi* | [**accountLevelCustomNameserversAddAccountCustomNameserver**](docs/AccountLevelCustomNameserversApi.md#accountLevelCustomNameserversAddAccountCustomNameserver) | **POST** /accounts/{identifier}/custom_ns | Add Account Custom Nameserver
*AccountLevelCustomNameserversApi* | [**accountLevelCustomNameserversDeleteAccountCustomNameserver**](docs/AccountLevelCustomNameserversApi.md#accountLevelCustomNameserversDeleteAccountCustomNameserver) | **DELETE** /accounts/{identifier}/custom_ns/{ns_name} | Delete Account Custom Nameserver
*AccountLevelCustomNameserversApi* | [**accountLevelCustomNameserversGetEligibleZonesForAccountCustomNameservers**](docs/AccountLevelCustomNameserversApi.md#accountLevelCustomNameserversGetEligibleZonesForAccountCustomNameservers) | **GET** /accounts/{identifier}/custom_ns/availability | Get Eligible Zones for Account Custom Nameservers
*AccountLevelCustomNameserversApi* | [**accountLevelCustomNameserversListAccountCustomNameservers**](docs/AccountLevelCustomNameserversApi.md#accountLevelCustomNameserversListAccountCustomNameservers) | **GET** /accounts/{identifier}/custom_ns | List Account Custom Nameservers
*AccountLevelCustomNameserversApi* | [**accountLevelCustomNameserversVerifyAccountCustomNameserverGlueRecords**](docs/AccountLevelCustomNameserversApi.md#accountLevelCustomNameserversVerifyAccountCustomNameserverGlueRecords) | **POST** /accounts/{identifier}/custom_ns/verify | Verify Account Custom Nameserver Glue Records
*AccountLevelCustomNameserversUsageForAZoneApi* | [**accountLevelCustomNameserversUsageForAZoneGetAccountCustomNameserverRelatedZoneMetadata**](docs/AccountLevelCustomNameserversUsageForAZoneApi.md#accountLevelCustomNameserversUsageForAZoneGetAccountCustomNameserverRelatedZoneMetadata) | **GET** /zones/{zone_identifier}/custom_ns | Get Account Custom Nameserver Related Zone Metadata
*AccountLevelCustomNameserversUsageForAZoneApi* | [**accountLevelCustomNameserversUsageForAZoneSetAccountCustomNameserverRelatedZoneMetadata**](docs/AccountLevelCustomNameserversUsageForAZoneApi.md#accountLevelCustomNameserversUsageForAZoneSetAccountCustomNameserverRelatedZoneMetadata) | **PUT** /zones/{zone_identifier}/custom_ns | Set Account Custom Nameserver Related Zone Metadata
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsCreateMonitor**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsCreateMonitor) | **POST** /accounts/{account_identifier}/load_balancers/monitors | Create Monitor
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsDeleteMonitor**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsDeleteMonitor) | **DELETE** /accounts/{account_identifier}/load_balancers/monitors/{identifier} | Delete Monitor
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsListMonitorReferences**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsListMonitorReferences) | **GET** /accounts/{account_identifier}/load_balancers/monitors/{identifier}/references | List Monitor References
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsListMonitors**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsListMonitors) | **GET** /accounts/{account_identifier}/load_balancers/monitors | List Monitors
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsMonitorDetails**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsMonitorDetails) | **GET** /accounts/{account_identifier}/load_balancers/monitors/{identifier} | Monitor Details
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsPatchMonitor**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsPatchMonitor) | **PATCH** /accounts/{account_identifier}/load_balancers/monitors/{identifier} | Patch Monitor
*AccountLoadBalancerMonitorsApi* | [**accountLoadBalancerMonitorsPreviewMonitor**](docs/AccountLoadBalancerMonitorsApi.md#accountLoadBalancerMonitorsPreviewMonitor) | **POST** /accounts/{account_identifier}/load_balancers/monitors/{identifier}/preview | Preview Monitor

docs/History.md
# History

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**alertBody** | **String** |  |  [optional]
**alertType** | **String** |  |  [optional]
**description** | **String** |  |  [optional]
**id** | **String** |  |  [optional]
**mechanism** | **String** |  |  [optional]
**mechanismType** | [**MechanismType**](MechanismType.md) |  |  [optional]
**name** | **String** |  |  [optional]
**sent** | [**OffsetDateTime**](OffsetDateTime.md) |  |  [optional]


src/main/java/io/swagger/client/model/Nft.java
/*
 * Cloudflare API
 * Interact with Cloudflare's products and services via the Cloudflare API. For more information on how to create and troubleshoot API tokens, refer to our [API fundamentals](https://developers.cloudflare.com/fundamentals/api/).  Totally new to Cloudflare? [Start here](https://developers.cloudflare.com/fundamentals/get-started/).
 *
 * OpenAPI spec version: 4.0.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
/**
 * Nft
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2023-02-22T13:40:27.702461541+09:00[Asia/Tokyo]")
public class Nft {
  @SerializedName("contract")
  private String contract = null;

  @SerializedName("token")
  private Integer token = null;

   /**
   * The ERC-721 compatible contract address.
   * @return contract
  **/
  @Schema(example = "0x57f1887a8bf19b14fc0d912b9b2acc9af147ea85", description = "The ERC-721 compatible contract address.")
  public String getContract() {
    return contract;
  }

   /**
   * The token ID for the NFT.
   * @return token
  **/
  @Schema(description = "The token ID for the NFT.")
  public Integer getToken() {
    return token;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Nft nft = (Nft) o;
    return Objects.equals(this.contract, nft.contract) &&
        Objects.equals(this.token, nft.token);
  }

  @Override
  public int hashCode() {
    return Objects.hash(contract, token);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Nft {\n");
    
    sb.append("    contract: ").append(toIndentedString(contract)).append("\n");
    sb.append("    token: ").append(toIndentedString(token)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
